packages/fetch-router/demos/node/app/posts.ts

```diff
@@ -0,0 +1,189 @@
+                      ${isCommentAuthor
+                        ? html`
+                            <form
+                              class="comment-delete-form"
+                              method="POST"
+                              action="${routes.posts.comment.destroy.href({
+                                ...getPostHrefParams(post),
+                                commentId: comment.id,
+                              })}"
+                            >
+                              <input type="hidden" name="_method" value="DELETE" />
+                              <button type="submit" class="btn-link btn-delete">Delete</button>
+                            </form>
+                          `
+                        : null}
+                    </div>
+                  `
+                })
+              : html`<p>No comments yet.</p>`}
+            ${currentUser
+              ? html`
+                  <form
+                    class="form-spaced"
+                    method="POST"
+                    action="${routes.posts.comment.create.href(getPostHrefParams(post))}"
+                  >
+                    <h3>Add a Comment</h3>
+                    <div class="form-field">
+                      <textarea name="content" rows="4" required></textarea>
+                    </div>
+                    <button type="submit">Post Comment</button>
+                  </form>
+                `
+              : html`<p><a href="${loginUrl(postUrl)}">Login</a> to add a comment.</p>`}
+          </section>
+        `,
+        currentUser,
+      ),
+    )
+  },
+  edit({ params, session }) {
+    let post = data.getPost(params.slug)
+    if (!post) {
+      return new Response('Post not found', { status: 404 })
+    }
+
+    let currentUser = session.get('username') as string | undefined
+    return res.html(templates.layout(postForm(post), currentUser))
+  },
+  update({ params, formData }) {
+    let post = data.getPost(params.slug)
+    if (!post) {
+      return new Response('Post not found', { status: 404 })
+    }
+
+    let title = formData.get('title') as string
+    let content = formData.get('content') as string
+
+    if (!title || !content) {
+      return res.html(templates.layout(postForm(post)), { status: 400 })
+    }
+
+    let updatedPost = data.updatePost(params.slug, title, content)
+    if (!updatedPost) {
+      return new Response('Post not found', { status: 404 })
+    }
+    let newSlug = generateSlug(title)
+    let finalPost = data.getPost(newSlug) ?? updatedPost
+    return res.redirect(routes.posts.show.href(getPostHrefParams(finalPost)))
+  },
+  destroy({ params }) {
+    let post = data.getPost(params.slug)
+    if (!post) {
+      return new Response('Post not found', { status: 404 })
+    }
+
+    data.deletePost(params.slug)
+    return res.redirect(routes.posts.index.href())
+  },
+} satisfies RouteHandlers<Omit<typeof routes.posts, 'comment'>>
```

Comment:
Using `satisfies` was necessary to get the types to cooperate when composing them all together in `router.ts`, so I opted for that approach whenever I used `RouteHandlers`